localstack/aws/handlers/response_aggregator.py

```diff
@@ -0,0 +1,64 @@
+import atexit
+import dataclasses
+import datetime
+import json
+import logging
+import threading
+from typing import Any, Dict
+
+from localstack.aws.api import RequestContext
+from localstack.aws.chain import HandlerChain
+from localstack.http import Response
+from localstack.utils import analytics
+from localstack.utils.scheduler import Scheduler
+
+LOG = logging.getLogger(__name__)
+FLUSH_INTERVAL_SECS = 10
+
+
+@dataclasses.dataclass
+class ResponseInfo:
+    service: str
+    operation: str
+    status_code: int
+
+    def to_string(self):
+        return json.dumps(dataclasses.asdict(self), sort_keys=True)
+
+
+class ResponseAggregator:
+    def __init__(self):
+        self.response_counts = {}
+        self.period_start_time = datetime.datetime.utcnow()
+        self.flush_scheduler = Scheduler()
+        self.scheduler_thread = threading.Thread(target=self.flush_scheduler.run)
+        self.scheduler_thread.start()
```

Comment:
we could use `start_worker_thread` here from the `localstack.utils.threads` package

Reply:
I attempted to drop in `start_worker_thread` this in but I get this error:
```TypeError: Scheduler.run() takes 1 positional argument but 2 were given```
Apparently `start_worker_thread` insists on passing `*args` and `**kwargs`. If you feel that `start_worker_thread` is the right thing to use then I'd be happy to go ahead and lightly refactor the scheduler and/or `start_worker_thread` to be more compatible - let me know what you think.

fwiw I based my implementation on the cloudwatch alarm scheduler:
https://github.com/localstack/localstack/blob/master/localstack/services/cloudwatch/alarm_scheduler.py#L42-L45